packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts

```diff
@@ -100,106 +96,162 @@ describe('UserController', () => {
   });
 
   it('throws error for POST /users with a string', async () => {
-    await client
+    const res = await client
       .post('/users')
       .send('hello')
       .expect(415);
+    expect(res.body.error.message).to.equal(
+      'Content-type application/x-www-form-urlencoded does not match [application/json].',
+    );
   });
 
   it('returns a user with given id when GET /users/{id} is invoked', async () => {
     const newUser = await userRepo.create(user);
     delete newUser.password;
     delete newUser.orders;
-    // MongoDB returns an id object we need to convert to string
-    // since the REST API returns a string for the id property.
-    newUser.id = newUser.id.toString();
 
     await client.get(`/users/${newUser.id}`).expect(200, newUser.toJSON());
   });
 
-  describe('user product recommendation (service) api', () => {
-    // tslint:disable-next-line:no-any
-    let recommendationService: HttpServer;
-
-    before(async () => {
-      recommendationService = createRecommendationServer();
-      await recommendationService.start();
-    });
+  describe('authentication', () => {
+    it('login returns a valid token', async () => {
+      await client
+        .post('/users')
+        .send(user)
+        .expect(200);
 
-    after(async () => {
-      await recommendationService.stop();
+      const res = await client
+        .post('/users/login')
+        .send({email: user.email, password: user.password})
+        .expect(200);
+
+      const token = res.body.token;
+      expect(token).to.not.be.Null();
+      expect(token).to.be.String();
+      expect(token).to.not.be.empty();
```

Comment:
The token should always be a string I think, it's enforced by https://github.com/strongloop/loopback4-example-shopping/blob/4fdc4b2ee2bfdc7369a8c0b67c3d329ed5d555b2/packages/shopping/src/controllers/user.controller.ts#L136